src/types.ts

```diff
@@ -2,3 +2,46 @@ import { PublicKey } from '@solana/web3.js';
 
 export type StringPublicKey = string;
 export type AnyPublicKey = StringPublicKey | PublicKey;
+
+export type MetaDataJsonCategory = 'image' | 'video' | 'audio' | 'vr';
+
+export type MetadataJsonAttribute = {
+  trait_type: string;
+  value: string;
+};
+
+export type MetadataJsonCollection = {
+  name: string;
+  family: string;
+};
+
+export type MetadataJsonFile = {
+  uri: string;
+  type: string;
+  cdn?: boolean;
+};
+
+export type MetadataJsonCreator = {
+  address: StringPublicKey;
+  verified: boolean;
+  share: number;
+};
+
+export type MetadataJsonProperties = {
+  files: MetadataJsonFile[];
+  category: MetaDataJsonCategory;
+  creators: MetadataJsonCreator[];
+};
+
+export type MetadataJson = {
```

Comment:
Pieced together from https://medium.com/metaplex/metaplex-metadata-standard-45af3d04b541
Does this look right?

Reply:
Ya seems reasonable.   Here is the token program metadata source of truth: https://github.com/metaplex-foundation/metaplex/blob/master/rust/token-metadata/program/src/state.rs#L88